cli/lsp/language_server.rs

```diff
@@ -1563,9 +1562,14 @@ impl Inner {
         }
         let reference_specifier =
           resolve_url(&reference.document_span.file_name).unwrap();
-        let asset_or_doc =
-          self.get_asset_or_document(&reference_specifier).await?;
-        results.push(reference.to_location(asset_or_doc.line_index(), self));
+        let reference_line_index = if reference_specifier == specifier {
```

Comment:
This also fixes a potential race condition where we re-use the line_index we first retrieved if the specifier is the same (which it is a lot of the time with this) instead of grabbing a new line_index.

Reply:
I think either way, there is no harm in re-using an already available value versus grabbing a new one.